### ISSUE TITLE
IDE0064: False positive when all readonly fields of a struct are static

### ISSUE THREAD

**Issue by airbreather** [dotnet/roslyn#57920]
**Version Used**: 
.NET SDK 6.0.100

**Steps to Reproduce**:
`dotnet build` the following:
### Repro.cs
```csharp
struct Repro
{
    public static readonly Repro DefaultValue = new();

    public int IrrelevantValue;

    public void Overwrite(Repro other) => this = other;
}
```

### Repro0.csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
  </PropertyGroup>

</Project>
```

### .editorconfig
```editorconfig
[*.cs]
dotnet_diagnostic.IDE0064.severity = warning
```
**Expected Behavior**:
IDE0064 is not raised: there are no **instance** fields that are declared `readonly`.

**Actual Behavior**:
IDE0064 is raised.

**Comment by Djoums** [comment]
Outside observer here, the current behavior seems right to me.
The warning is here to prevent a readonly field from changing value if the struct instance is reassigned, because that would break the readonly contract for an outside observer. But in the case of static fields that doesn't apply, since they're independant of any instance.
